webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.hooks.js

```diff
@@ -0,0 +1,14 @@
+import { useState, useContext } from 'react';
```

Comment:
+1 for creating a separate file for hooks. IMO we should call it `ForemanModalHooks.js` to align with the other conventions.